### PR TITLE
Support macOS in KMT scenarios

### PR DESCRIPTION
**Pull request by gjulianm**
What does this PR do?
---------------------

Enable support for KMT native VMs in macOS, with the folllowing changes

- Changing the hypervisor type to hvf in macOS 
- Relying on QEMU support of native virtualized networking in macOS (vmnet), instead of libvirt as it doesn't support it.
- Domain IPs must be discovered dynamically from DHCP leases in macOS, this requires changing the type of the `domain.ip` variable to `pulumi.StringOutput`.
- Updates pulumi-libvirt 
- Add a test.check-xslt task to easily test XSL transformations.
- Change the path of the libvirt socket on macOS
- Add remote/local working directories to the configuration, to allow for macOS hosts to launch both local and remote VMs.
- Enable the NFS connection killer when the host system is macOS (see https://github.com/DataDog/ami-builder/pull/161)

Which scenarios this will impact?
-------------------

KMT scenarios

Motivation
----------

Support for local KMT VMs in macOS

Additional Notes
----------------

### CODE REVIEW

File: tasks/test.py
Comment (usamasaqib) [inline tasks/test.py:1]: Really nice!

File: scenarios/aws/microVMs/microvms/run.go [hunk 1]
```diff
@@ -59,11 +64,15 @@ var SSHKeyFileNames = map[string]string{
...
+// GetWorkingDirectory is a function that returns the working directory for the kernel version testing, given the architecture (local or non-local).
+var GetWorkingDirectory func(string) string
 
-func getKernelVersionTestingWorkingDir(m *config.DDMicroVMConfig) func() string {
-	return func() string {
-		return m.GetStringWithDefault(m.MicroVMConfig, config.DDMicroVMWorkingDirectory, "/tmp")
+func getKernelVersionTestingWorkingDir(m *config.DDMicroVMConfig) func(string) string {
+	return func(arch string) string {
+		if arch == LocalVMSet {
+			return m.GetStringWithDefault(m.MicroVMConfig, config.DDMicroVMLocalWorkingDirectory, "/tmp")
```
[the remark this answers is not in the document]
Comment (gjulianm) [reply scenarios/aws/microVMs/microvms/run.go:73]: We could check that the return value is not "/tmp", the problem is that we don't have `m.GetString` functions, seems like everything is geared towards not erroring

File: scenarios/aws/microVMs/microvms/network.go [hunk 2]
```diff
@@ -167,3 +170,80 @@ func generateNetworkResource(
...
+
+	return leases, nil
+}
+
+// waitForBootpDHCPLeases waits for the macOS DHCP server (BootP) to assign an IP address to the VM based on its MAC address, and
+// returns that IP address.
+func waitForBootpDHCPLeases(mac string) (string, error) {
+	// The DHCP server will assign an IP address to the VM based on its MAC address, wait until it is assigned
+	// and then return the IP address.
+	for {
```
[the remark this answers is not in the document]
Comment (gjulianm) [reply scenarios/aws/microVMs/microvms/network.go]: Added a 5 minute timeout

**Review by usamasaqib — APPROVED** [review]
Minor comment. Otherwise looks good.

File: scenarios/aws/microVMs/microvms/run.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (usamasaqib) [reply scenarios/aws/microVMs/microvms/run.go:73]: You can always add the `GetString` function if you like

File: scenarios/aws/microVMs/microvms/network.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (usamasaqib) [reply scenarios/aws/microVMs/microvms/network.go]: Maybe dump the dhcp leases if the timeout is reached. Might help debug issues.

**Pull request merged**